File: FactoryMethod/LoggerFactory.py
from FileLogger import *
from ConsoleLogger import *
from DBLogger import *

class LoggerFactory():
    """Class that contains factory method."""
    def getLogger(self,x):
        if x==1:
            return FileLogger()
        elif x==2:
            return ConsoleLogger()
        elif x==3:
            return DBLogger()

    

File: Decorator/Messages/LoggerFactory.java
import java.util.*;
import java.io.*;

public class LoggerFactory {

  public boolean isFileLoggingEnabled() {
    Properties p = new Properties();
    try {
      p.load(ClassLoader.getSystemResourceAsStream(
        "Logger.properties"));
      String fileLoggingValue =
        p.getProperty("FileLogging");
      if (fileLoggingValue.equalsIgnoreCase("ON") == true)
        return true;
      else
        return false;
    } catch (IOException e) {
      return false;
    }
  }

  public Logger getLogger() {
    if (isFileLoggingEnabled()) {
      return FileLogger.getFileLogger();
    } else {
      return new ConsoleLogger();
    }
  }

}
